src/electron/utils/storage.js

```diff
@@ -54,6 +58,49 @@ class Database {
         return validator
     }
 
+    async checkUniqueValidatorIds(validatorIds, stakingAddress) {
+        const collidingIds = []
+        const obj = this._store.get('stakerAddress')
+        if (!obj) return true
+        for (const validatorId of validatorIds) {
+            if (obj[stakingAddress]
+                && obj[stakingAddress].validators
+                && obj[stakingAddress].validators[validatorId]) {
+                collidingIds.push(validatorId)
+            }
+        }
+        if (collidingIds.length > 0) {
+            await this.checkPhasesOfCollidingIds(collidingIds).catch(e => {
+                console.log(e.message)
+                throw new Error(e.message)
+            })
+        }
+        return true
+    }
+
+    async checkPhasesOfCollidingIds(collidingIds) {
+        console.log("checking phases of colliding ids", collidingIds)
+        const ALCHEMY_API_KEY = 'Wo1ZTJHKM_EqFK5Q7weYMNeeR42VShY3'
+        const ALCHEMY_URL = `https://eth-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`; 
+        const NODE_MANAGER_ADDRESS = '0x8b71140ad2e5d1e7018d2a7f8a288bd3cd38916f';
+        const web3=new Web3(ALCHEMY_URL);
+        const validatorsInWrongPhase = []
+        const contract = new web3.eth.Contract(ABI, NODE_MANAGER_ADDRESS);
```

Comment:
can we move this out to not be initialized every time this function runs ? should speed things up quick a bit